hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -78,7 +78,7 @@ public class DatanodeAdminMonitorImpl implements DatanodeAdminMonitor {
   private ReplicationManager replicationManager;
   private Queue<DatanodeDetails> pendingNodes = new ArrayDeque();
   private Queue<DatanodeDetails> cancelledNodes = new ArrayDeque();
-  private Set<DatanodeDetails> trackedNodes = new HashSet<>();
+  private Map<String, TrackedNode> trackedNodes = new HashMap<>();
```

Comment:
Why does this need to be changed to a Map, rather than Set? The trackNode class has hashCode and equals based only on DatanodeDetails, so we could keep this as a Set. Maybe there is a reason you need to change it to a map I have missed?

Reply:
I have changed Set<DatanodeDetails> trackedNodes to Map<String, TrackedNode> trackednodes as in the future, we will want to fetch information of a particular datanode -> this would be difficult if we store it as a set as there is no get method (worst case, we'll need to iterate through the whole set and find a match).

Reply:
On second thoughts, keeping it as set will be better as most changes we are making to the tracked nodes are when we are iterating through the set. Moreover, even if we wish to fetch a particular DN, the set won't be very big so iterating through it to find one should be fine. I have reverted this back to set.